src/sql/base/browser/ui/table/asyncDataView.ts

```diff
@@ -126,8 +130,9 @@ export class VirtualizedCollection<T extends Slick.SlickData> implements IObserv
 		let currentData = this.getRangeFromCurrent(start, end);
 
 		// only shift window and make promise of refreshed data in following condition:
-		if (start < this._bufferWindowBefore.getStartIndex() || end > this._bufferWindowAfter.getEndIndex()) {
+		if (this._lengthChanged || start < this._bufferWindowBefore.getStartIndex() || end > this._bufferWindowAfter.getEndIndex()) {
```

Comment:
Does it make more sense to just reset the window around the new length when a new length is set, if we have a window that includes the old length The way this is written doesn't seem like it functionally makes sense.

Reply:
The length may change several times while a query is running and this makes us only pay the cost to rebuild the windows once when actually needed.  Not sure what doesn't make sense, but if you have an alternate implementation could you please send a PR with the proposed fix?  The problem is we are getting windows created that don't contain a full window of data since the length is changing and the calculations in this code block are incorrect in that scenario.